pkg/sdk/tags.go

```diff
@@ -1,18 +1,152 @@
 package sdk
 
-// placeholder for the real implementation.
-type TagCreateOptions struct{}
+import (
+	"context"
+	"database/sql"
+	"strings"
+	"time"
+)
+
+type Tags interface {
+	Create(ctx context.Context, request *CreateTagRequest) error
+	Show(ctx context.Context, opts *ShowTagRequest) ([]Tag, error)
+	ShowByID(ctx context.Context, id AccountObjectIdentifier) (*Tag, error)
+	Drop(ctx context.Context, request *DropTagRequest) error
+	Undrop(ctx context.Context, request *UndropTagRequest) error
+}
+
+// createTagOptions is based on https://docs.snowflake.com/en/sql-reference/sql/create-tag
+type createTagOptions struct {
+	create        bool                    `ddl:"static" sql:"CREATE"`
+	OrReplace     *bool                   `ddl:"keyword" sql:"OR REPLACE"`
+	tag           string                  `ddl:"static" sql:"TAG"`
+	IfNotExists   *bool                   `ddl:"keyword" sql:"IF NOT EXISTS"`
+	name          AccountObjectIdentifier `ddl:"identifier"`
```

Comment:
Shouldn't it be SchemaObjectIdentifier? (based on https://docs.snowflake.com/en/user-guide/object-tagging#what-is-a-tag and our current implementation, i.e. `createTagWithOptions`)

Reply:
fixed